metronome/config/specs/src/io/iohk/metronome/config/ConfigParserSpec.scala

```diff
@@ -53,6 +55,72 @@ class ConfigParserSpec
     json.noSpaces shouldBe """{"nestedStructure":{"barBaz":{"spam":"eggs"},"foo":10}}"""
   }
 
+  "seq,list,vector decoders" should "handle JSON Object with sequential indices successfully" in {
+    import ConfigParserSpec.TestConfigWithJsonArray._
+    val withArray  = """{"field":["valueA", "valueB", "valueC"]}"""
+    val withObject = """{"field":{"2":"valueC", "0": "valueA", "1":"valueB"}}"""
+    val expected   = Vector("valueA", "valueB", "valueC")
+    val aConf      = ConfigFactory.parseString(withArray)
+    val oConf      = ConfigFactory.parseString(withObject)
+
+    // seqs
+    val aSeq = ConfigParser.parse[TestConfigWithSeq](aConf.root)
+    val oSeq = ConfigParser.parse[TestConfigWithSeq](oConf.root)
+    inside(aSeq) { case Right(value) => value.field.toVector shouldBe expected }
+    aSeq shouldEqual oSeq
+
+    // lists
+    val aList = ConfigParser.parse[TestConfigWithList](aConf.root)
+    val oList = ConfigParser.parse[TestConfigWithList](oConf.root)
+    inside(aList) { case Right(value) =>
+      value.field.toVector shouldBe expected
+    }
+    aList shouldEqual oList
+
+    // vectors
+    val aVector = ConfigParser.parse[TestConfigWithVector](aConf.root)
+    val oVector = ConfigParser.parse[TestConfigWithVector](oConf.root)
+    inside(aVector) { case Right(value) => value.field shouldBe expected }
+    aVector shouldEqual oVector
+  }
+
+  "seq,list,vector decoders" should "fail on JSON Object key gaps" in {
+    import ConfigParserSpec.TestConfigWithJsonArray._
+    // format: off
+    val gapsConf1 = ConfigFactory.parseString("""{"field":{"2":"valueC", "0":"valueA"}}""")
+    val gapsConf2 = ConfigFactory.parseString("""{"field":{"2":"valueA"}}""")
+
+    checkDecoding[TestConfigWithSeq](gapsConf1, Left(s"Expected [0, 2) sequence, but got {0, 2}"))
+    checkDecoding[TestConfigWithSeq](gapsConf2, Left(s"Expected [0, 1) sequence, but got {2}"))
+
+    checkDecoding[TestConfigWithList](gapsConf1, Left(s"Expected [0, 2) sequence, but got {0, 2}"))
+    checkDecoding[TestConfigWithList](gapsConf2, Left(s"Expected [0, 1) sequence, but got {2}"))
+
+    checkDecoding[TestConfigWithVector](gapsConf1, Left(s"Expected [0, 2) sequence, but got {0, 2}"))
+    checkDecoding[TestConfigWithVector](gapsConf2, Left(s"Expected [0, 1) sequence, but got {2}"))
+    // format: on
+  }
+
+  "seq,list,vector decoders" should "succeed on empty JSON Object" in {
+    import ConfigParserSpec.TestConfigWithJsonArray._
+    // format: off
+    val emptyConf = ConfigFactory.parseString("""{"field":{}}""")
+    checkDecoding[TestConfigWithSeq](emptyConf, Right(_.field shouldBe empty))
+    checkDecoding[TestConfigWithList](emptyConf, Right(_.field shouldBe empty))
+    checkDecoding[TestConfigWithVector](emptyConf, Right(_.field shouldBe empty))
+    // format: on
+  }
+
+  "seq,list,vector decoders" should "succeed on JSON Object with duplicated keys keeping the last value" in {
```

Comment:
Thank you for making an explicit test case for duplicated keys. 

I wanted to say that it would be more helpful to reject such input, but I suppose by the time we get a JSON object `circe` has already collapsed it into `Map` and we can't tell that there were any duplicate values, right?